From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Mon, 10 Jul 2023 22:05:08 +0200
Subject: [PATCH] Modifiable Bukkit enums - Inject runtime versions - Add
 module opens

Add module opens, to both the server JAR and to development runs,
that are needed to perform internal operations on the system class loader.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/build.gradle.kts b/build.gradle.kts
index e21c5979b993e6d39c681ac2762250d6825a1605..958ac76821c872f791186bf8defd477f714e37e9 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -87,6 +87,7 @@ tasks.jar {
         val gitBranch = git("rev-parse", "--abbrev-ref", "HEAD").getText().trim() // Paper
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
+            "Add-Opens" to "java.base/java.lang", // Fiddle - modifiable Bukkit enums - inject runtime versions - add module opens to server jar
             "Implementation-Title" to "CraftBukkit",
             "Implementation-Version" to "git-Fiddle-$implementationVersion", // Fiddle - branding changes - rename server brand
             "Implementation-Vendor" to date, // Paper
@@ -196,6 +197,7 @@ fun TaskContainer.registerRunTask(
     val memoryGb = providers.gradleProperty("paper.runMemoryGb").getOrElse("2")
     minHeapSize = "${memoryGb}G"
     maxHeapSize = "${memoryGb}G"
+    jvmArgs("--add-opens=java.base/java.lang=ALL-UNNAMED") // Fiddle - modifiable Bukkit enums - inject runtime versions - add module opens to development runs
 
     doFirst {
         workingDir.mkdirs()
